fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/recordrepair/RecordValidateAndRepairTest.java

```diff
@@ -444,39 +378,50 @@ void testValidateRecordCombinationSplitMissing(FormatVersion formatVersion, BitS
             commit(context);
         }
 
-        RecordRepairRunner.ValidationKind validationKind = RecordRepairRunner.ValidationKind.RECORD_VALUE_AND_VERSION;
+        RecordRepair.ValidationKind validationKind = RecordRepair.ValidationKind.RECORD_VALUE_AND_VERSION;
         FDBRecordStore.Builder storeBuilder;
         try (FDBRecordContext context = openContext()) {
             final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
             storeBuilder = store.asBuilder();
         }
-        RecordRepairRunner runner = RecordRepairRunner.builder(fdb).build();
-        List<RecordValidationResult> repairResults = runner.runValidationAndRepair(storeBuilder, validationKind, true);
-
-        Map<Integer, RecordValidationResult> validationResultMap = repairResults.stream()
-                .collect(Collectors.toMap(res -> (int)res.getPrimaryKey().getLong(0), res -> res));
-
-        // Assert that both records are either gone or are valid or flagged as corrupt
-        Assertions.assertThat(
-                        ValidationTestUtils.recordWillDisappear(2, splitsToRemove, formatVersion) ||
-                                ValidationTestUtils.recordWillRemainValid(2, splitsToRemove, formatVersion) ||
-                                validationResultMap.containsKey(ValidationTestUtils.RECORD_ID_WITH_TWO_SPLITS))
-                .isTrue();
-
-        Assertions.assertThat(
-                        ValidationTestUtils.recordWillDisappear(3, splitsToRemove, formatVersion) ||
-                                ValidationTestUtils.recordWillRemainValid(3, splitsToRemove, formatVersion) ||
-                                validationResultMap.containsKey(ValidationTestUtils.RECORD_ID_WITH_THREE_SPLITS))
-                .isTrue();
-
+        RecordRepair.Builder builder = RecordRepair.builder(fdb, storeBuilder).withValidationKind(validationKind);
+        try (RecordRepairValidateRunner runner = builder.buildRepairRunner(true)) {
+            // Run validate and repair
+            RepairValidationResults repairResults = runner.run().join();
+
+            int validResults = NUM_RECORDS;
+            if (ValidationTestUtils.recordWillDisappear(2, splitsToRemove, formatVersion)) {
```

Comment:
It is unclear what `2` and `3` are referring to here. Perhaps putting them in variables at the top of the method would help?

Reply:
Kind of, it may also be more readable with intelliJ, if it is filling in `primaryKey:` for the arguments, so this looks more like:
```
if (ValidationTestUtils.recordWillDisappear(primaryKey: 2, splitsToRemove, formatVersion)) {
```

But it feels like a bit of a juqmp to the fact that 2 is the primary key for a record with 2 splits, and 3 is the primary key for a record with 3 splits.